Finished ToDoList Project

Dandush03 opened the pull request:
# To Do List

This project allowed you to keep a nice track of the stuff that you have to do and help you visualize it's priorities

Dandush03 committed:
Added Main Structure to project with esLint web pack and Sass linter

Dandush03 committed:
Fixing linter errors

Dandush03 committed:
Fixing linter errors

Dandush03 committed:
added default values and filled main content

Dandush03 committed:
fixing linter errors

Dandush03 committed:
fixing linter errors

Dandush03 committed:
added form functionalties

Dandush03 committed:
added delete functions to the row

Dandush03 committed:
added delete functions to the row

Dandush03 committed:
Fixing git Ignore

Dandush03 committed:
fixed frm id issue

Dandush03 committed:
styled left menu and added fonts

Dandush03 committed:
added new frm

Dandush03 committed:
added linter roles

Dandush03 committed:
deleted sass file

Dandush03 committed:
added sass file

Dandush03 committed:
added sass rules

Dandush03 committed:
added sass rules

Dandush03 committed:
added sass rules

Dandush03 committed:
testing sass rules

Dandush03 committed:
testing sass rules

Dandush03 committed:
testing sass rules

Dandush03 committed:
testing sass rules

Dandush03 committed:
testing sass rules

Dandush03 committed:
fixed linter errors

Dandush03 committed:
fixed linter errors

Dandush03 committed:
Fixing Linter errors

Dandush03 committed:
Fixing Linter errors

Dandush03 committed:
Fixing Linter errors

Dandush03 committed:
Finished Project

Dandush03 committed:
fixed minor bug

Dandush03 committed:
added some efect

Dandush03 committed:
Fixed Linter Errors

Torres-ssf commented on package.json:
```diff
@@ -0,0 +1,31 @@
+  "description": "4th Microverse JS Project",
+  "private": true,
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
```
Since no tests are being implemented, we can safely remove this script.

Torres-ssf commented on package.json:
```diff
@@ -0,0 +1,31 @@
+  "private": true,
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "build": "webpack"
```
It's a good practice to add more scripts to your package file. You can create at least 2 scripts, one for productions(build) and another one to the development stage(dev). It can be something like:
```
    "dev": "webpack --mode development",
    "build": "webpack --mode production",
```

Torres-ssf reviewed (changes requested):
### Changes required :muscle: 
Hi, there @Dandush03, how are you?
Great job in this project :clap:. I can see that you put a lot of work on it :nerd_face: 
But before moving on there are some things that we need to change.

- [x] It is good that you have informed all the technologies used in your application in the description of the PR. But a better idea would be to leave this for the README file, where all additional information about your application resides. And in the PR description, you should summarize all changes made to the branch. Like what features were implemented. Please use this [article](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243642/) created by Microverse as a guide.

- [x] As it said in the instructions, we should leave the app logic separated from the DOM manipulation. This means that DOM manipulations tasks should happen in a different file from your app logic.
![Screenshot from 2020-04-24 08-20-18](https://user-images.githubusercontent.com/30977845/80207419-79392200-8604-11ea-9345-68f7c9dc6c04.png)

- [x] It seems that right now there isn't a way to mark a todo task as complete/finished. The Status section isn't showing anything.

- [x]  The todo priority can't bee is seen after creating the todo. The only way to see it is when you click to edit the todo.

**[OPTIONAL]**

- [x] It makes more sense to set the input type of priority as a Radio. So the user can only choose between 3  options, low, medium, and high.

Please submit for code review again after making these changes.
Feel free to message me on slack if you have any questions about this review.

Happy coding :computer: 
**Sergio Torres**

Dandush03 committed:
Added Functionalities requested by TSE

Dandush03 committed:
fixed package.json

Dandush03 commented:
# Hey! :wave: 

### All done, Thanks for your review

mcrd25 commented on src/index.js:
```diff
@@ -0,0 +1,55 @@
+};
+
+/*
+function local() {
+  Object.keys(localStorage).forEach(i => {
+    localStorage.removeItem(i);
+  });
+}  */
```
Please remove commented out code.
 It’s better to delete commented code. If you need to review a previous version of your code, use git history
Please see this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from pathwright

mcrd25 commented on src/index.js:
```diff
@@ -0,0 +1,55 @@
+}  */
+
+window.onload = () => {
+  // local();
```
Please remove commented out code 🙏

mcrd25 reviewed (changes requested):
# Changes Required 👀 🔍 

Hi Team! 👋 

You did a good job implementing your changes, however there are two key issues I still find:
- You still have not Addressed Application logic/ DOM Manipulation separation.  You ought to have application logic (no DOM manipulation like creating elements, changing contents of elements etc) in separate module and DOM manipulation in separate module. This is not currently the case. You can see logic.js module actually has both Application Logic and DOM manipulation.
- You forgot to remove commented out code

Hope this review helps. 🙂 

***Please makes changes and submit another review request.
If you have any questions, let me know here or on slack and I will be glad to clarify with you. You can also reach me and connect with me on my social media or email.***

*Happy coding 😄,*
**[--Maya](https://github.com/mcrd25)**
[GitHub](https://github.com/mcrd25) [Twitter](https://twitter.com/mcrd25) [linkedin](https://www.linkedin.com/in/mayadouglas/) [GitLab](https://gitlab.com/mcrd25)

---
*P.S. all requested changes not marked by [OPTIONAL] or are not positive comments are **REQUIRED***

Dandush03 committed:
fixed logic removed coments.

Dandush03 commented:
## Hey again, 

#### I deleted all if statement from index file and separate al DOM element from it. if this doesn't suit your need please be more specified, since on my last commit fallowing the other review I removed all functions from all DOM files. 

#### Deleted all commented code.

## Thanks for your code review, looking forward for your replay

shloch commented on src/views/javascripts/logic.js:
```diff
@@ -0,0 +1,240 @@
+  document.body.removeChild(mainDiv);
+}
+
+function closeTaskFrm() {
+  const main = document.getElementById('mainContainer');
+  const container = document.getElementsByClassName('task-popup');
+  main.removeChild(container[0]);
+}
```
this is a sample of what we mean by manipulating the DOM inside a logic code
(You practically named this file `logic.js` yourself)

shloch commented on src/views/javascripts/logic.js:
```diff
@@ -0,0 +1,240 @@
+  };
+};
+
+function closeFrm(mainDiv) {
+  document.body.removeChild(mainDiv);
+}
+
```
this is another DOM manipulation

shloch commented on src/views/javascripts/logic.js:
```diff
@@ -0,0 +1,240 @@
+  const inputs = frm.querySelectorAll('input, textarea, date, time, select');
+  let div = document.getElementById('error-msg');
+
+  if (div === null) {
+    div = document.createElement('div');
+    div.className = 'error-msg';
+    div.id = 'error-msg';
+  } else {
+    div.innerHTML = '';
+  }
```
this is DOM manipulation

shloch commented on src/views/javascripts/logic.js:
```diff
@@ -0,0 +1,240 @@
+  inputs.forEach(element => {
+    const { value, name } = element;
+    if (value === '') {
+      const span = document.createElement('span');
+      span.innerText = `${name} can't be empty`;
+      div.appendChild(span);
```
this is DOM manipulation

shloch commented on src/views/javascripts/logic.js:
```diff
@@ -0,0 +1,240 @@
+  let classN = parentNode.className;
+  classN = classN.split(' ');
+  let newClass = '';
+  classN.forEach(classes => {
+    if (classes === 'close') {
+      newClass += 'open';
+      this.innerHTML = '<i class="fas fa-chevron-up"></i>';
+    } else if (classes === 'open') {
+      newClass += 'close';
+      this.innerHTML = '<i class="fas fa-chevron-down"></i>';
+    } else {
+      newClass += `${classes} `;
+    }
```
this is DOM manipulation

shloch reviewed (changes requested):
## hello @Dandush03 ,
I have not gone very deep into your code to understand everything probably like the other TSEs, but i must confess your user interface is not user-friendly. Todo's ought to be inside `PROJECTS`. Maybe your UI has it, but i just couldn't find it.

a project is something like "COOKING", and TASKS things like
- cut tomatoes
- grind spices
- chop carrots to pieces.

Now let's forget all i said above. Now below, i have pointed out some of the DOM-related code you've included inside your `logic.js` which is what the other TSEs are pointing out.

In a Logic file, there should only be pure data manipulation ( creation/modification/deletion), with factories, objects, arrays, activeStorage  etc....  At the end, this data manipulated is passed to a DOM code (most likely another code in another file in charge of managing the DOM).

There's shouldn't be Manipulation of DOM at all inside your Logic. You might just `READ` from the DOM in order to obtain data which you will manipulate but not manipulate DOM itself (by adding elements, changing styles etc) like you have done below:

I just pointed out some (not all)  below:

Please, work on that

Dandush03 committed:
Modified Logic.Js and deleted all DOM manipulation on it

Dandush03 committed:
deleted comented code

Dandush03 commented:
### Hey :wave:
#### Hope you're having a nice day,
@shloch, i'm very sorry to hear that you didn't and I quote -"not gone very deep into your code to understand everything", there most be a web incompatibilities issue since my page look like this on my side:
![Screenshot from 2020-04-23 16-48-40](https://user-images.githubusercontent.com/51087936/80427503-e32c2280-88b5-11ea-93b2-44a79252e56c.png)
and in any case i recheck my code to make sure it was displaying and also asked some of my stand up team to check it out. as you can see from this image and in the README file the ToDoList is in the left corner.

I fixed my logic.js File and deleted all DOM manipulation from it as required. Thanks for your review.

## Looking forward for any other comment regarding my code. 
# Happy Coding Everyone :computer:

bellom reviewed (changes requested):
## Hi @Dandush03  :clap: 

**Great work making `almost` all the necessary corrections from the previous review.**
 **Logic separated from DOM 👍** 

Update your `README` with correct information; 
Example: you're using the live link from `tictactoe` project 😞 

**Please Note:** The only reason why you will need to send in another re-review request is due to the fact that I can't add new tasks to project created, watch the video below;
Kindly share with us if we are missing something, thank you.

**Kindly watch this [video code review](https://www.loom.com/share/188a56d62ae8468d9351f9ef32682562);**



**Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....**

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

Dandush03 committed:
fixed Chrome adaptation

Dandush03 committed:
fixing linter errors

Dandush03 commented:
# Hey @bellom  :wave: 
### Thanks for that great video. Now I understand what was the issue.
basically I was using FireFox and everything was looking and working good on my side, but when I saw your video you where using Chrome, and when I switch I saw all does bug and manage to fix them.

## Thanks one more time for your review. It was really helpful

### Have a nice day! :wink:

bellom reviewed (approved):
## Hi @Dandush03  :clap: 

**Great work making all the necessary corrections from the previous review.**

**_I trust you can fix the little bug before merging; your task `priority` should be a dropdown feature_**

## Congratulation :+1:  
_Go ahead and merge to dev and don't forget to submit project completion form.
Good luck in the next project :slightly_smiling_face:_ 

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**
